### DISCUSSION TITLE
Adding google map as base layer

### DISCUSSION THREAD

**Discussion by akhilvs000** [Mapsui/Mapsui#1460] [Q&A]
Is it possible to add google map as base map tile in MapsUI.
If yes, could someone share a sample

**Comment by pauldendulk** [comment]
The short answer is no. Google requires you to use there map component to use their data. Also their api protocol is very complicated in some parts. 

However, in BruTile there is sample for the ordinary map and the terrain (but not the aerial photos). If you want to use them you need to contact google (one of our users actually did this and made an agreement, but it was a long time ago).
- https://github.com/BruTile/BruTile/blob/master/Samples/BruTile.Samples.Common/Samples/GoogleMapsSample.cs
- https://github.com/BruTile/BruTile/blob/master/Samples/BruTile.Samples.Common/Samples/GoogleMapsTerrainSample.cs

For aerial photos you could use Bing. Look at:
https://github.com/Mapsui/Mapsui/blob/master/Samples/Mapsui.Samples.Common/Maps/Demo/BingSample.cs
You need an api key. I am not sure if you need to use their map component. You should contact them.
